src/components/Layout/NetworkSelect/NetworkSelect.tsx

```diff
@@ -0,0 +1,178 @@
+  onClick?: () => void;
+}
+
+interface DropdownItemLabelProps {
+  network: ChainInfoObject;
+  className?: string;
+  active?: boolean;
+}
+
+/**
+ * Dropdown control for the network selection
+ */
+const WrappedDropdown = (props: DropdownProps) => {
+  const { children, className, ...rest } = props;
+  return (
+    <div className={className}>
+      <Dropdown className="rounded-md py-1 pl-4 p-2 border border-gray-300 bg-white" {...rest}>
+        {children}
+      </Dropdown>
+    </div>
+  );
+};
+
+const StyledDropdown = styled(WrappedDropdown)`
+  display: inline-block;
+  min-width: 12.5rem;
+  font-size: 0.87rem;
+
+  span.select-msg {
+    padding: 0.25rem 0.7rem;
+    color: darkgrey;
+    cursor: default;
+  }
+`;
+
+/**
+ * Label for the items of the dropdown list
+ */
+const DropdownItemLabel: FunctionComponent<DropdownItemLabelProps> = ({ className, active, network }) => {
+  return (
+    <div className={className} data-testid={`network-select-dropdown-label-${network.chainId}`}>
+      <img className="network-icon" src={network.iconImage} alt={network.label} />
+      <span className="label">{network.label}</span>
+      {active ? <span className="active" /> : null}
+    </div>
+  );
+};
+
+const StyledDropdownItemLabel = styled(DropdownItemLabel)`
+  display: flex;
+  align-items: center;
+
+  & img.network-icon {
+    width: 20px;
+    height: 20px;
+    border-radius: 20px;
+    margin-right: 0.5rem;
+  }
+
+  & span.label {
+    width: 100%;
+  }
+
+  & span.active {
+    padding: 0.25rem;
+    background-color: #27ae60;
+    border-radius: 0.5rem;
+    justify-self: flex-end;
+  }
+`;
+
+/**
+ * Dropdown item label specially for unsupported networks
+ */
+const DropdownUnsupportedLabel = styled.div`
+  color: #bb2323;
+
+  &::before {
+    content: "❌";
+    margin-right: 0.5rem;
+  }
+`;
+
+/**
+ * Item component for the dropdown list
+ */
+const NetworkSelectDropdownItem = (props: NetworkSelectDropdownItemProps) => {
+  const { className, network, active, ...rest } = props;
+  return (
+    <div className={className}>
+      <DropdownItem {...rest}>
+        <StyledDropdownItemLabel network={network} active={active} />
+      </DropdownItem>
+    </div>
+  );
+};
+
+/**
+ * Network Selection dropdown component
+ */
+const NetworkSelectView: FunctionComponent<NetworkSelectViewProps> = ({ onChange, networks, currentChainId }) => {
+  const itemsList = networks.map((network, i) => {
+    return (
+      <NetworkSelectDropdownItem
+        key={i}
+        network={network}
+        active={network.chainId === currentChainId}
+        onClick={() => {
+          if (onChange) onChange(network);
+        }}
+      />
+    );
+  });
+
+  let selectedLabel: React.ReactNode = <DropdownUnsupportedLabel>Unsupported Network</DropdownUnsupportedLabel>;
+  try {
+    if (currentChainId) {
+      selectedLabel = <StyledDropdownItemLabel network={getChainInfo(currentChainId)} />;
+    }
+  } catch (_) {}
```

Comment:
interesting, what the underscore means? like intentionally not gonna handle the error?

Reply:
Oops. I think I initially had something in the catch then I removed it and forgotten about this. Updated in 14e486a5

Reply:
mmm perhaps a console log on the error will do? not sure if that's right. iirc try not to swallow errors

Reply:
Sure, added console log in c93da8d4